From b3477096afcccbd1a502d4378ddf2122f0bda3ec Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Tue, 6 Jun 2017 10:59:44 -0700
Subject: [PATCH] IMGAPI-630 ExportImage called by admin (sdc-imgadm export)
 should be able to export to other account's Manta area

---
 docs/index.md | 2 +-
 lib/images.js | 9 +++------
 2 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/docs/index.md b/docs/index.md
index c3dc4a8..d3d1f4a 100644
--- a/docs/index.md
+++ b/docs/index.md
@@ -1319,7 +1319,7 @@ in Manta can be exported, locally stored images are not supported.
 
 | Field                 | Type   | Required? | Notes                                                                                                                                                                                                                                                                           |
 | --------------------- | ------ | --------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| account (query param) | UUID   | No\*      | The account UUID on behalf of whom this request is being made. If given then the manta_path prefix must resolve to a location that is owned by the account. If not given then the manta_path prefix is assumed to (and must) resolve to a path that is owned by the admin user. |
+| account (query param) | UUID   | No\*      | The account UUID on behalf of whom this request is being made. If given then the manta_path prefix must resolve to a location that is owned by the account. |
 | channel (query param) | String | No        | The image channel to use. (Only relevant for servers using [channels](#channels).)                                                                                                                                                                                              |
 | manta_path            | String | Yes\*     | Manta path prefix where the image file and manifest should be exported to. If "manta_path" is a dir, then the files are saved to it. If the basename of "PATH" is not a dir, then "PATH.imgmanifest" and "PATH.zfs[.EXT]" are created.                                          |
 
diff --git a/lib/images.js b/lib/images.js
index 6673217..8d2ffa6 100644
--- a/lib/images.js
+++ b/lib/images.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright 2016 Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 /*
@@ -4552,12 +4552,9 @@ function apiExportImage(req, res, callback) {
             exportImage();
         });
     // account is not given:
-    //      Call from admin user, can only export to storage.manta.user
+    //      This is a call by the admin user. IMGAPI places *no guard* on
+    //      where in Manta it will export.
     } else {
-        if (muser !== stor.user) {
-            return callback(
-                new errors.NotMantaPathOwnerError(stor.user, mpath));
-        }
         exportImage();
     }
 
-- 
2.21.0

